pyomo/contrib/iis/mis.py

```diff
@@ -0,0 +1,382 @@
+        return msg
+
+    results = solver.solve(modified_model, tee=tee)
+    if pyo.check_optimal_termination(results):
+        msg = _constraint_loop("variable bounds", msg)
+
+    # next, try relaxing the inequality constraints
+    for v in slack_block.component_data_objects(pyo.Var):
+        c = _get_constraint(modified_model, v)
+        if c.equality:
+            # equality constraint
+            continue
+        if v not in fixed_slacks:
+            v.unfix()
+
+    results = solver.solve(modified_model, tee=tee)
+    if pyo.check_optimal_termination(results):
+        msg = _constraint_loop("inequality constraints and/or variable bounds", msg)
+
+    for v in slack_block.component_data_objects(pyo.Var):
+        if v not in fixed_slacks:
+            v.unfix()
+
+    results = solver.solve(modified_model, tee=tee)
+    if pyo.check_optimal_termination(results):
+        msg = _constraint_loop(
+            "inequality constraints, equality constraints, and/or variable bounds", msg
+        )
+
+    if len(elastic_filter) == 0:
+        # load the feasible solution into the original model
+        for modified_model_var, v in _modified_model_var_to_original_model_var.items():
+            v.set_value(modified_model_var.value, skip_validation=True)
+        results = solver.solve(model, tee=tee)
+        if pyo.check_optimal_termination(results):
+            logger.info(f"A feasible solution was found!")
+        else:
+            logger.info(
+                f"Could not find a feasible solution with violated constraints or bounds. This model is likely unstable"
+            )
+
+    # Phase 2 -- deletion filter
+    # TODO: the model created here seems to mess with the nl_v2
+    #       writer sometimes. So we temporarily switch to nl_v1 writer.
+    WriterFactory.register("nl")(WriterFactory.get_class("nl_v1"))
+
+    # remove slacks by fixing them to 0
+    for v in slack_block.component_data_objects(pyo.Var):
+        v.fix(0)
+    for o in modified_model.component_data_objects(pyo.Objective, descend_into=True):
+        o.deactivate()
+
+    # mark all constraints not in the filter as inactive
+    for c in modified_model.component_data_objects(pyo.Constraint):
+        if c in elastic_filter:
+            continue
+        else:
+            c.deactivate()
+
+    try:
+        results = solver.solve(modified_model, tee=tee)
+    except:
+        results = None
+
+    if pyo.check_optimal_termination(results):
+        msg += "Could not determine Minimal Intractable System\n"
+    else:
+        deletion_filter = []
+        guards = []
+        for constr in elastic_filter:
+            constr.deactivate()
+            for var, val in _modified_model_value_cache.items():
+                var.set_value(val, skip_validation=True)
+            try:
+                results = solver.solve(modified_model, tee=tee)
+            except:
+                math_failure = True
+            else:
+                math_failure = False
```

Comment:
You could set `math_failure = False` before this `try` block so you don't need an `else`